### PR TITLE
Updated Status command to use threads

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
Updated `metal status` command to use threads. This removed all the single handling and most of the kill commands. The `bash` commands still operate on their own `sub processes` however each process has it's own `Job` thread managing it in rube.

Also using the inbuilt `Queue` class in ruby and removed the old `Jobs` class. The old `Task` class has been renamed as `Job` to make the naming consistent

### CODE REVIEW

**Comment by WilliamMcCumstie** [comment]
Also added test which makes the bulk of the additions

**Comment by BobWhitelock** [comment]
Rebased this on latest `develop` for easier reviewing.

File: lib/ruby/lib/alces/stack/status/run.rb [hunk 1]
```diff
@@ -19,138 +19,116 @@
...
+          while data["FINISHED"] != true
+            data = get_finished_data
+            if data.empty?
+              empty_count += 1
+              raise DataIncomplete if empty_count > 100 && @monitor.thread.stop?
+            elsif data["FINISHED"] != true
+              display_data data
+              empty_count = 0
+            end
+            sleep 0.1
```
Comment (BobWhitelock) [inline lib/ruby/lib/alces/stack/status/run.rb]: Why the magic sleep? Could be worth adding a comment to this with the reason as the purpose isn't obvious, and similarly for other magic sleeps.

**Comment by BobWhitelock** [comment]
@mjtko are you happy for this to be merged once @WilliamMcCumstie has looked at my comments, or do you want to have a look through yourself first?

**Comment by WilliamMcCumstie** [comment]
Will merge the changes in with the new PR #50.

Once the current PR are merged into `feature/refactored-status` I will make the changes as described by @BobWhitelock

File: lib/ruby/lib/alces/stack/status/run.rb [hunk 2]
```diff
@@ -19,138 +19,116 @@
...
+      end
+
+      def nodes
+        @nodes ||= _set_nodes
+      end
+
+      def _set_nodes
+        lambda_proc = lambda { |lambda_hash| lambda_hash[:nodename] }
+        a = Alces::Stack::Iterator.run(group, lambda_proc, nodename: nodename)
+        a.is_a?(Array) ? a : [a]
```
Comment (WilliamMcCumstie) [inline lib/ruby/lib/alces/stack/status/run.rb]: Will need to look at the updated version of the iterator

Comment (BobWhitelock) [inline lib/ruby/lib/alces/stack/status/run.rb]: I think this is irrelevant/resolved with the new `Nodes` class, don't worry about this

File: lib/ruby/lib/alces/stack/status/run.rb [hunk 3]
```diff
@@ -19,138 +19,116 @@
...
+    end
+
     module Status
       class Run
-        include Alces::Tools::Execution
-
         def initialize(options={})
-          @opt = Options.new(options)
+          @opt = RunOptions.new(options)
+          raise "Can not specify both -g and -n" if @opt.nodename? == @opt.group?
```
Comment (WilliamMcCumstie) [inline lib/ruby/lib/alces/stack/status/run.rb]: So the condition is correct it's the error message that is miss leading. It is intended as general input validation of the two

**Pull request closed without merging**